src/ansys/fluent/core/utils/search.py

```diff
@@ -224,10 +248,429 @@ def inner(tree, path, root_path):
                     next_path = f'{path}.{k}["<name>"]'
                 else:
                     next_path = f"{path}.{k}"
+                type_ = "Object" if isinstance(v, Mapping) else v
+                api_object_names.add(k)
+                if "tui" in next_path:
+                    api_tui_objects.append(f"{next_path} ({type_})")
+                else:
+                    api_objects.append(f"{next_path} ({type_})")
                 if _match(k, word, match_whole_word, match_case):
-                    type_ = "Object" if isinstance(v, Mapping) else v
                     print(f"{next_path} ({type_})")
             if isinstance(v, Mapping):
                 inner(v, next_path, root_path)
 
     inner(api_tree, "", root_path)
+
+    def _write_api_tree_file(
+        api_objects: list,
+        name: Optional[bool] = None,
+        text: Optional[bool] = False,
+        tui: Optional[bool] = False,
+    ):
+        api_tree_file = get_api_tree_file_name(text=text, name=name, tui=tui)
+        api_tree_file.touch()
+        with open(api_tree_file, "w") as file:
+            for api_object in api_objects:
+                file.write(f"{api_object}")
+                file.write("\n")
+
+    text_file_folder = Path(os.path.join(pyfluent.CODEGEN_OUTDIR, "api_tree"))
+    text_file_folder.mkdir(parents=True, exist_ok=True)
+
+    _write_api_tree_file(api_objects=api_objects, text=True)
+    _write_api_tree_file(api_objects=api_tui_objects, tui=True)
+    _write_api_tree_file(api_objects=api_object_names, name=True)
+
+
+def _get_api_object_names():
```

Comment:
This function is called multiple times instead of being called once and the resultant list being passed down. Those lower-level functions should be independent of the API file so that they might be fully testable.

Reply:
Yes, we are going to write a single file so that we will load it once and use everywhere. Thank you.

Reply:
Done. Thank you.